app/nexus/internal/state/state.go

```diff
@@ -130,12 +130,16 @@ func UpsertSecret(path string, values map[string]string) {
 //   - path: The path to the secret to be deleted
 //   - versions: A slice of version numbers to delete. If empty, deletes the
 //     current version only. Version number 0 is the current version.
-func DeleteSecret(path string, versions []int) {
+func DeleteSecret(path string, versions []int) error {
 	kvMu.Lock()
-	kv.Delete(path, versions)
 	kvMu.Unlock()
 
-	persist.AsyncPersistSecret(kv, path)
+	err := kv.Delete(path, versions)
+	if err != nil {
+		return err
+	} // TODO ? THIS RETURN BREAKS THE CODE FLOW ASK IS IT CORRECT ?
+
+	return persist.AsyncPersistSecret(kv, path)
```

Comment:
This is an async operation; it should not be part of the return.

`persist.AsyncPersistSecret()` should NOT have a return value as it is a background operation.

Reply:
^ I take it back.. there is a `getrawsecret()` in that call which may return an error.

Reply:
I removed return then logged error in here